app/services/katello/pulp3/repository/generic.rb

```diff
@@ -30,6 +36,63 @@ def self.api(smart_proxy, repo)
           api_class = RepositoryTypeManager.find_by(:pulp3_service_class, self).pulp3_api_class
           api_class ? api_class.new(smart_proxy, repo) : Katello::Pulp3::Api::Core.new(smart_proxy)
         end
+
+        def create_distribution(path)
+          distribution_data = api.class.distribution_class(repo.repository_type).new(secure_distribution_options(path))
+          api.distributions_api.create(distribution_data)
+        end
+
+        def create_remote
+          remote_file_data = api.class.remote_class(repo.repository_type).new(remote_options)
+          response = api.remotes_api.create(remote_file_data)
+          repo.update!(:remote_href => response.pulp_href)
+        end
+
+        def refresh_distributions
+          if repo.docker?
```

Comment:
You don't need the check for `docker?` since docker isn't going to be a generic type.